### PR TITLE
Add Build Flow view for better visualization of Upstream and Downstream Builds

### LINKED ISSUE #1310: Add Build Flow view: interactive graph of upstream/downstream build relationships
### What feature do you want to see added?

A Build Flow view that visualizes upstream and downstream build-trigger relationships as an interactive directed graph (DAG). This brings the core value of [Yet Another Build Visualizer](https://github.com/jenkinsci/yet-another-build-visualizer-plugin) directly into Pipeline Graph View with a modern implementation using the Jenkins Design Library.

### What it does

- Renders pipeline builds as a directed graph with bezier-curve edges connecting trigger-linked builds
- Zoom, pan, and fit-to-view (mouse wheel, trackpad pinch, drag, or button controls)
- Toggle upstream/downstream visibility independently
- Switch between left-to-right DAG layout and top-to-bottom layout
- Flatten the graph to a simple grid for large pipelines
- Toggle display of: duration, build numbers, full job names, descriptions
- Shows up to 6 recent build result dots per node (click to navigate)
- Auto-refreshes every 5s while any build is in-progress
- Focus-path highlighting on hover (dims unrelated nodes/edges)
- Staggered entrance animations for nodes and edges
- Safety caps: MAX_NODES=200, MAX_DEPTH=50

### Where it appears

- **Build page overview tab**: Embedded card above Stages
- **Build sidebar**: "Build Flow" link opens a dedicated full-page view
- **Job page sidebar**: Shows the latest build's flow

### Detection method

Trigger-based only:
- Upstream: `CauseAction` / `Cause.UpstreamCause`
- Downstream: `DownstreamBuildAction`
- Queued items: linked through `Cause.UpstreamCause`

### Are you interested in contributing this feature?

Yeah, can work on it in the free time :)

### PR DESCRIPTION
**Pull request by adityajalkhare**
Closes #1310

## Summary

Adds a Build Flow view that renders upstream and downstream build-trigger relationships as an interactive, zoomable directed graph. Uses the shared Jenkins Design Library status icons, Tippy.js dropdowns, and react-zoom-pan-pinch for pan/zoom.

#### Screenshot
image

#### Live Video
https://github.com/user-attachments/assets/486caeb3-ce9a-47da-ba7d-a50f266b834d

#### Focused View of a Job
https://github.com/user-attachments/assets/1829b836-1e7f-4d5a-ab61-ffc91c100412

#### Build Flow Tab
bbuild-flow-tab

#### Overview Tab
overview-tab

#### Job Page
job-page

## Features

- Directed graph (DAG) layout with bezier-curve edges
- Zoom, pan, fit-to-view
- Toggle upstream/downstream builds independently
- Switch between LTR and TTB layout directions
- Flatten to grid mode for large pipelines
- Focus-path highlighting on hover
- Staggered entrance animations for nodes and edges
- Display toggles: duration, build number, full names, description
- Up to 6 recent result dots per node (clickable)
- Auto-refresh (5s) while any build is in-progress
- Safety caps: 200 nodes, 50 depth
- Full i18n support
- Accessibility: aria labels, reduced-motion, high-contrast, forced-colors

## Where it appears

| Context | View |
|---------|------|
| Build overview tab | Embedded card (350px height) |
| Build sidebar | Full-page dedicated view |
| Job sidebar | Latest build's flow |

## Detection method

Trigger-based only. Uses `Cause.UpstreamCause` and `DownstreamBuildAction`. Does not scan artifacts or copyArtifacts relationships.

## Implementation

| Layer | Files | Role |
|-------|-------|------|
| Frontend | `BuildFlow.tsx`, `BuildFlowLayout.ts`, `BuildFlowUtils.ts`, `BuildFlow.scss` | React component, DAG layout algorithm, utilities, styling |
| Model | `BuildFlowModel.ts`, `BuildFlowNode.java`, `BuildFlowEdge.java` | Shared types for the graph response |
| Backend | `BuildFlowGraph.java`, `BuildFlowResponse.java` | Traverses build chain, computes DAG, serializes JSON |
| Actions | `BuildFlowAction.java`, `BuildFlowJobAction.java` + factories | Sidebar links, API endpoint at `buildFlow/graph` |
| Views | 6 Jelly files | Renders in new + legacy Jenkins UI modes |

## Testing

- **Frontend**: 214 tests across 27 files (vitest) - layout, utilities, SCSS tokens, focus-path, status mapping
- **Backend**: 7 integration tests (@WithJenkins) - graph traversal, edge cases, parallel/nested stages
- **Manual**: Verified with `mvn hpi:run` on sequential, parallel, nested-stage, and multi-branch pipelines

### CODE REVIEW

**Comment by KalleOlaviNiemitalo** [comment]
Does this detect upstream/downstream relationships solely from how builds are **triggered**, or does this also detect if a downstream build is triggered by something unrelated (e.g. [cron](https://www.jenkins.io/doc/book/pipeline/syntax/#triggers)) and then uses the [copyArtifacts](https://www.jenkins.io/doc/pipeline/steps/copyartifact/#copyartifacts-copy-artifacts-from-another-project) step to copy **artifacts** from an upstream build?

I'm asking because we can get hundreds of downstream builds in the copyArtifacts case and I'm concerned that looking up all of those could be excessively slow.

**Comment by adityajalkhare** [comment]
@KalleOlaviNiemitalo 

Thanks for calling this out. I checked the implementation to make sure before replying.

This feature is trigger-based only, not artifact-based.

- Upstream detection uses `CauseAction` / `Cause.UpstreamCause` in `getUpstreamBuild(...)`.
- Downstream detection uses `DownstreamBuildAction` in `getDownstreamBuilds(...)`.
- Queued downstream items are also linked only through `Cause.UpstreamCause`.
- There is no `copyArtifacts`-specific detection logic in this feature.

So if a build starts independently, for example from cron, and then uses `copyArtifacts` from some other build, this implementation would **not** treat that build as downstream.

So the graph is following explicit build-trigger relationships, not artifact-consumption relationships.

On the performance concern, this code is not scanning all builds looking for artifact usage. It reads trigger metadata from the current build chain, and it also has hard safety caps:

- `MAX_NODES = 200`
- `MAX_DEPTH = 50`

So the case of getting hundreds of downstream builds purely because of `copyArtifacts` should not apply here.

That said, I do not have an exact or near-replica of your pipeline setup locally, so if you have a representative case handy, please feel free to test it. If it behaves differently in practice, I can follow up and revert or adjust the implementation as needed.

**Comment by novinxy** [comment]
Looks awesome! 😍
Does it work when upstream triggers downstream with build step ?

**Comment by timja** [comment]
I used this to re-create your screenshot:

image


```groovy
import com.cloudbees.hudson.plugins.folder.Folder
import org.jenkinsci.plugins.workflow.job.WorkflowJob
import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition
import jenkins.model.Jenkins

// ---------------------------------------------------------------------------
// Recreates the upstream/downstream "Build Flow" pipeline jobs under issues/1310.
//
// Wiring (upstream -> downstream). Where a job has several upstreams it gets
// triggered once per upstream, which is why e2e-tests and deploy-staging show
// up as multiple runs in the Build Flow view.
// ---------------------------------------------------------------------------

def downstream = [
    'root'              : ['frontend-build', 'backend-build', 'mobile-build'],
    'frontend-build'    : ['frontend-unit-test', 'frontend-lint'],
    'backend-build'     : ['backend-unit-test', 'backend-lint', 'backend-security'],
    'mobile-build'      : ['mobile-unit-test'],
    'frontend-unit-test': ['e2e-tests'],
    'frontend-lint'     : ['e2e-tests'],
    'backend-unit-test' : ['integration-tests'],
    'backend-lint'      : ['integration-tests'],
    'backend-security'  : ['integration-tests'],
    'mobile-unit-test'  : ['mobile-e2e'],
    'e2e-tests'         : ['deploy-staging'],
    'integration-tests' : ['deploy-staging'],
    'mobile-e2e'        : [],
    'deploy-staging'    : [],
]

def jenkins = Jenkins.get()
def rng = new Random()

// --- ensure folder path issues/1310 ---------------------------------------
def getOrCreateFolder = { ItemGroup parent, String name ->
    def existing = parent.getItem(name)
    if (existing instanceof Folder) {
        return existing
    }
    if (existing != null) {
        existing.delete() // wrong type, replace it
    }
    parent.createProject(Folder, name)
}

def issues = getOrCreateFolder(jenkins, 'issues')
def folder = getOrCreateFolder(issues, '1310')

// --- build the pipeline script for a job -----------------------------------
def scriptFor = { String name, List<String> kids ->
    def secs = 2 + rng.nextInt(7) // 2..8s so the tree completes quickly
    def sb = new StringBuilder()
    sb << "pipeline {\n"
    sb << "  agent any\n"
    sb << "  stages {\n"
    sb << "    stage('${name}') {\n"
    sb << "      steps {\n"
    sb << "        echo 'Running ${name}'\n"
    sb << "        sleep time: ${secs}, unit: 'SECONDS'\n"
    sb << "      }\n"
    sb << "    }\n"
    if (kids) {
        sb << "    stage('trigger downstream') {\n"
        sb << "      steps {\n"
        kids.each { kid ->
            // relative name resolves within the same folder
            sb << "        build job: '${kid}', wait: false, propagate: false\n"
        }
        sb << "      }\n"
        sb << "    }\n"
    }
    sb << "  }\n"
    sb << "}\n"
    sb.toString()
}

// --- create / update each job ---------------------------------------------
downstream.each { name, kids ->
    def job = folder.getItem(name)
    if (!(job instanceof WorkflowJob)) {
        if (job != null) {
            job.delete()
        }
        job = folder.createProject(WorkflowJob, name)
    }
    job.definition = new CpsFlowDefinition(scriptFor(name, kids), true)
    job.save()
    println "Configured ${job.fullName}"
}

println "\nDone. Created/updated ${downstream.size()} jobs under ${folder.fullName}."
println "Trigger the flow with:  Jenkins.get().getItemByFullName('issues/1310/root').scheduleBuild2(0)"

// Uncomment to kick off the whole tree immediately:
// folder.getItem('root').scheduleBuild2(0)

```

File: src/main/frontend/build-flow-view/build-flow/main/BuildFlow.scss [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+}
+
+.pgv-build-flow {
+  position: relative;
+  height: 350px;
+  overflow: hidden;
+
+  // Full-page view: fill available space
+  &--full-page {
+    height: calc(100vh - 300px);
```
Comment (timja) [inline src/main/frontend/build-flow-view/build-flow/main/BuildFlow.scss]: any reason not to do 100% ?

Seems to be not using the space the best when its on its own page?

**Comment by adityajalkhare** [comment]
@novinxy - yes. I used that only.
image

File: src/main/frontend/build-flow-view/build-flow/main/BuildFlow.scss (hunk 1 above)
Comment (adityajalkhare) [reply src/main/frontend/build-flow-view/build-flow/main/BuildFlow.scss]: There seemed to be some issues with the spacing in initial iterations, but I just tested it with 100% and it works fine now.

Have updated the code, and also added a feature for better UX - non-connecting curves/lines get dimmed on hover of a node.

**Comment by timja** [comment]
I haven’t looked in detail yet and will need more time and ideally feedback from others
Some initial thoughts:
* the pipeline you are actively viewing (how it adds dashed lines) feels a bit strange to me. It took me a bit to work out why it was dashed I think this could be communicated differently.
* I’m not sure on the second widget for it on the build and job pages. Maybe there should be a button for swapping between the two widgets? I get it makes it more discoverable but we don’t want lots of widgets piled on that page as it will get full quickly

**Comment by adityajalkhare** [comment]
@timja 

> the pipeline you are actively viewing (how it adds dashed lines) feels a bit strange to me. It took me a bit to work out why it was dashed I think this could be communicated differently.

Thanks, I did think about it but left it for community feedback because couldn't decide. The following is another design which Jenkins follows widely -

image

WDYT about this?

**Comment by timja** [comment]
I'm not sure if the others need fading out but yes some sort of border to show 'active' or 'selected' would be better

**Comment by adityajalkhare** [comment]
The fade out was not on select, it was on hover... I just took a screenshot at the wrong time :)

**Comment by novinxy** [comment]
For me fade out looks clear and makes it easy to see selected "flow" 
But "active" border is also nice alternative :)

Those builds on each card are stages from the pipelines ?
Do they provide links to the stages ?
What happens if there are many stages like 100 in pipeline ?

_Edit: Nvm, description answered that for me. Those are recent builds_

**Comment by novinxy** [comment]
This flow shows pipeline relations, but is it easy to analyze individual flow of the builds?

I mean such situation:
- we have 3 pipelines: A, B, C
- A -> B -> C
- they already have more than 5 runs
- Scenario A: Pipeline A fails without triggering B. Last Build for A, B, C is "different build flow"
- Scenario B: Pipeline A fails but triggered B, which triggered B, C. Last Build for A, B, C is "same build flow"
- Is there option to differenciate scenario A and scenario B from those graphs ? Is it possible to track which build from pipeline A triggered which build from pipeline B?

If not then I think simple "active border" on each related run would be sufficient :)
 So If I hover on one run, it will "active" and other cascade or runs will also active :)

**Comment by adityajalkhare** [comment]
Just for clarification -

### Selected/Active border on current job
image

### Hover fade out (can still see the current job in the left side, but faded)
image

**Comment by novinxy** [comment]
> Just for clarification -
> 
> ### Selected/Active border on current job
> image
> ### Hover fade out (can still see the current job in the left side, but faded)
> image

Maybe hover effect should be disabled if it hides active pipeline ?

But with clicking on active card maybe we could enhance it better ? What if clicking could bring out pipeline flow? That could help in analyzing flow for complicated scenarios, e.g.
Click on one pipeline shows fades out all non-related pipelines, or changes border for all related ones
If we click on frontend-lint we make "green ones" more visible
image

what do you think about this idea ?
Sorry if I'm proposing something really difficult, I'm not frontend guy, so I don't know how difficult are such implementations 😅

**Comment by adityajalkhare** [comment]
> Scenario A: Pipeline A fails without triggering B. Last Build for A, B, C is "different build flow"

The build flow does not load in this case for the latest build where A failed before triggering B.

> Scenario B: Pipeline A fails but triggered B, which triggered B, C. Last Build for A, B, C is "same build flow"

If A fails after triggering B, which triggered C, and if `wait: true` was set, that means C was finished, then B, then some steps ran which failed, so the Build Flow would load.

_We can also test these scenarios for better understanding. _

> Maybe hover effect should be disabled if it hides active pipeline ?

Hover is intended for a _quick glance_ of the flow between job nodes (one parent and one child), while being anywhere in the graph, irrespective of the selected/active job.

> But with clicking on active card maybe we could enhance it better ? What if clicking could bring out pipeline flow? That could help in analyzing flow for complicated scenarios, e.g.
Click on one pipeline shows fades out all non-related pipelines, or changes border for all related ones
If we click on frontend-lint we make "green ones" more visible

The though process, and what I have seen is - Bezier curves/Connector lines are the tracking mechanisms users use to track the flow.

I would prefer not fading the whole unrelated flow based on selected job, unless there is a strong reasoning since then the "here's the whole Build Flow" core idea and glancing through functionality would break.

When I think of it..... There is an hide/show upstream/downstream toggle, perhaps there can be another one, "View Active Job Flow/name something else" which can let us view what you suggested. @novinxy WDYT about this?

> Sorry if I'm proposing something really difficult, I'm not frontend guy, so I don't know how difficult are such implementations 😅

I'm also just starting out, so no idea lol... ;)

**Comment by adityajalkhare** [comment]
@timja @novinxy 
https://github.com/user-attachments/assets/1829b836-1e7f-4d5a-ab61-ffc91c100412

Also updated it in description

File: src/main/java/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowJobActionFactory.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import hudson.model.Job;
+import java.util.Collection;
+import java.util.Collections;
+import jenkins.model.TransientActionFactory;
+
+/**
+ * Registers {@link BuildFlowJobAction} on every {@link Job}.
+ */
+@Extension
+public class BuildFlowJobActionFactory extends TransientActionFactory<Job> {
```
Comment (timja) [inline src/main/java/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowJobActionFactory.java:15]: do we need this on a job?

Watching it while running it was surprising that it rebuilt the flow for the running build i.e. the last flow disappeared and as new jobs started they appeared in the flow.

as currently built it seems more build specific, you can always turn on the previous build options when viewing it to see how its been running over time.

File: README.md [hunk 3]
```diff
@@ -30,6 +31,25 @@ This plugin adds a visual representation of Jenkins pipelines, showing each stag
 
 Hidden steps are not displayed by default in the Pipeline Overview, but can be toggled visible using the filter controls.
 
+## Build Flow
```
Comment (timja) [inline README.md]: can you move this after the screenshots section and include a screenshot of this too

File: src/main/java/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowGraph.java [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+        if (result.equals(Result.FAILURE)) {
+            return "FAILURE";
+        }
+        if (result.equals(Result.UNSTABLE)) {
+            return "UNSTABLE";
+        }
+        if (result.equals(Result.ABORTED)) {
+            return "ABORTED";
+        }
+        return "NOT_BUILT";
```
Comment (timja) [inline src/main/java/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowGraph.java]: This is simpler, (If we bump Jenkins version we could move to java 21 to simplify further)

```suggestion
        if (run.isBuilding()) {
            return "IN_PROGRESS";
        }

        Result result = run.getResult();
        if (result == null) {
            return "IN_PROGRESS";
        }

        return switch (result.toString()) {
            case "SUCCESS", "FAILURE", "UNSTABLE", "ABORTED" -> result.toString();
            default -> "NOT_BUILT";
        };
```

File: src/main/java/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowGraph.java [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+    }
+
+    private static List<String> getRecentResults(@NonNull Run<?, ?> run) {
+        List<String> results = new ArrayList<>(6);
+        // Include the current build's result as the most recent entry
+        results.add(mapStatus(run));
+        Run<?, ?> previous = run.getPreviousBuild();
```
Comment (timja) [inline src/main/java/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowGraph.java]: We should avoid loading previous builds into memory if they aren't already there.

This is how JUnit does it:

https://github.com/jenkinsci/junit-plugin/blob/5c8bbfb9798f39a5d31ab95785c1da5e5fcea45e/src/main/java/hudson/tasks/test/AbstractTestResultAction.java#L262-L292

File: src/main/frontend/build-flow-view/build-flow/main/BuildFlow.tsx [hunk 6]
```diff
@@ -0,0 +1,957 @@
...
+const MAX_SCALE = 3;
+const CONTROLS_OVERHEAD = 80; // px reserved for zoom/toggle overlays
+const MIN_CONTAINER_HEIGHT = 150;
+const MAX_CONTAINER_HEIGHT = 350;
+
+const STORAGE_KEY = "pgv-build-flow-prefs";
+
+// --- Preferences persistence ---
+
+interface BuildFlowPrefs {
```
Comment (timja) [inline src/main/frontend/build-flow-view/build-flow/main/BuildFlow.tsx]: don't shorten names, applies to all occurrences:
```suggestion
interface BuildFlowPreferences {
```

File: src/main/frontend/build-flow-view/build-flow/main/BuildFlow.tsx [hunk 7]
```diff
@@ -0,0 +1,957 @@
+import "./BuildFlow.scss";
```
Comment (timja) [inline src/main/frontend/build-flow-view/build-flow/main/BuildFlow.tsx:1]: Can you refactor this file to split some sections of it out logically, e.g. components or grouped functions

its a large file as-is

File: src/main/frontend/build-flow-view/build-flow/main/BuildFlowIcons.tsx [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+      r="32"
+      fill="none"
+      stroke="currentColor"
+      strokeMiterlimit="10"
+      strokeWidth="32"
+    />
+  </svg>
+);
+
+export const IconClose = (
```
Comment (timja) [inline src/main/frontend/build-flow-view/build-flow/main/BuildFlowIcons.tsx:202]: an example of what should be in a different file, isn't there an existing close icon you can reuse though?

File: src/main/resources/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowAction/widget.jelly [hunk 9]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core">
+  <j:if test="${it.shouldDisplayBuildFlow()}">
+    <div class="jenkins-card"
```
Comment (timja) [inline src/main/resources/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowAction/widget.jelly]: Ideally this widget would be integrated into the existing stages widget with a button to switch between them.

Maybe an option to enable the second widget but not enabled by default.

Widgets and new tabs 

Image

should be added conservatively so that we don't end up with loads of them cluttering the page

**Review by timja — COMMENTED** [review]
looking really good, I've had a reasonable look over although there's a lot of code.
few comments for improvements

File: src/main/resources/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowAction/widget.jelly (hunk 9 above)
Comment (adityajalkhare) [reply src/main/resources/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowAction/widget.jelly]: I thought and propose renaming the Stages widget to Flows, keeping the same icon for a wider coverage of both Stage Flows and Build Flows, WDYT?

File: src/main/java/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowJobActionFactory.java (hunk 2 above)
Comment (adityajalkhare) [reply src/main/java/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowJobActionFactory.java:15]: I added this on the job page since that is where a user "lands" first and can easily look through the flow for the latest build, finished or currently running, and browse upstream/downstream without going on the build page (for the latest build at least).

This behavior is the same as how it is in [Yet Another Build Visualizer](https://github.com/jenkinsci/yet-another-build-visualizer-plugin).

From as user perspective, I would like it there.

From the perspective we are discussing it, I suggest we give users a toggle in Global Settings to Show/Hide Build Flow like we have for Stage Names and Duration.

WDYT?

File: src/main/java/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowGraph.java (hunk 4 above)
Comment (adityajalkhare) [reply src/main/java/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowGraph.java]: So we move to baseline Jenkins 2.555.3 then?

File: src/main/resources/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowAction/widget.jelly (hunk 9 above)
Comment (adityajalkhare) [reply src/main/resources/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowAction/widget.jelly]: I have removed the Build Flow Tab and integrated the jelly for it with `PipelineConsoleViewAction/widget.jelly`.

However, it will render the overview page as follows -
image


To actually move Build Flow below Console, we'd need to either make `BuildFlowAction` a `Tab` action again so its widget renders on the `Overview` without disturbing `Stages` and `Console`.

My proposal to rename the `Stages` Tab as `Flows` or something else that can be discussed still stands.

File: src/main/java/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowJobActionFactory.java (hunk 2 above)
Comment (adityajalkhare) [reply src/main/java/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowJobActionFactory.java:15]: I have added a toggle in Global Settings which ADMINS can use to enable/disable the Build Flow on Job Page.

For a USER specific toggle, I am thinking where to place it.

Comment (adityajalkhare) [reply src/main/java/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowJobActionFactory.java:15]: The USER toggle -

image

File: src/main/resources/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowAction/widget.jelly (hunk 9 above)
Comment (timja) [reply src/main/resources/io/jenkins/plugins/pipelinegraphview/buildflow/BuildFlowAction/widget.jelly]: my suggestion was to combine them into one widget and allow switching between them.

i.e. Add a build flow button to the stages one

**Comment by panicking** [comment]
@adityajalkhare ant plan to continue?

**Comment by adityajalkhare** [comment]
@panicking yeah. I'll pick it up after 15th July and see if I can add it to YABV's own repo if it doesn't take a lot of time.

Last I saw it had lot's of upgrade issues open.

If not I'll see if creating a new plugin in itself is a better idea.

But the Build Flow won't be a part of Pipeline Graph View as others suggested above.

**Comment by panicking** [comment]
> @panicking yeah. I'll pick it up after 15th July and see if I can add it to YABV's own repo if it doesn't take a lot of time.
> 
> Last I saw it had lot's of upgrade issues open.
> 
> If not I'll see if creating a new plugin in itself is a better idea.
> 
> But the Build Flow won't be a part of Pipeline Graph View as others suggested above.

I think the flow should go here, I don't see any reason why this should not happen. There use case where the jobs are composed by multiple pipeline? Why the flow as a tab is not welcome?

**Comment by adityajalkhare** [comment]
@panicking, @janfaracik and @timja have strongly suggested to keep this flow in the OG plugin itself in #1310

**Comment by panicking** [comment]
> @panicking, @janfaracik and @timja have strongly suggested to keep this flow in the OG plugin itself in #1310

@adityajalkhare Are you continue to working on this patch?
